src/main/java/com/cgdpd/library/validation/Validator.java

```diff
@@ -33,6 +34,26 @@ public static String requiredValidIsbn(String paramName, String value) {
         return value;
     }
 
+    public static Optional<Long> checkTrackingStatus(String paramName,
+                                                     TrackingStatus trackingStatus,
+                                                     Optional<Long> value) {
+        switch (trackingStatus) {
```

Comment:
Your IDE should be highlighting this
`Switch statement can be replaced with enhanced 'switch' `
Make sure to read them as they are usually good suggestions.
With the enhanced switch the compiler guarantees that we are checking all possibilities, if not it will throw a compiler error